api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1248,9 +1248,78 @@ const (
 // +kubebuilder:validation:Enum=Karpenter
 type Provisioner string
 
+// Configures when and how to scale down cluster nodes.
+type ScaleDownConfig struct {
+	// enabled determines whether the Cluster Autoscaler should scaled down this cluster.
+	//
+	// +kubebuilder:default=Enabled
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +optional
+	Enabled string `json:"enabled,omitempty"`
+
+	// delayAfterAddSeconds sets how long after scale up the scale down evaluation resumes in seconds
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=10000
+	// +kubebuilder:default=60
+	// +optional
+	DelayAfterAddSeconds *int32 `json:"delayAfterAddSeconds,omitempty"`
+
+	// delayAfterDelete sets how long after node deletion, scale down evaluation resumes, defaults to scan-interval
+	// +kubebuilder:validation:Pattern=([0-9]*(\.[0-9]*)?[a-z]+)+
+	// +kubebuilder:validation:MaxLength=100
+	// +optional
+	DelayAfterDelete *string `json:"delayAfterDelete,omitempty"`
+
+	// delayAfterFailure sets how long after a scale down failure, scale down evaluation resumes
+	// +kubebuilder:validation:Pattern=([0-9]*(\.[0-9]*)?[a-z]+)+
+	// +kubebuilder:validation:MaxLength=100
+	// +optional
+	DelayAfterFailure *string `json:"delayAfterFailure,omitempty"`
+
+	// unneededDurationSeconds establishes how long a node should be unneeded before it is eligible for scale down in seconds
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=10000
+	// +kubebuilder:default=600
+	// +optional
+	UnneededDurationSeconds *int32 `json:"unneededDurationSeconds,omitempty"`
+
+	// utilizationThreshold determines the node utilization level, defined as sum of requested resources divided by capacity, below which a node can be considered for scale down
+	// +kubebuilder:validation:Pattern=(0.[0-9]+)
+	// +kubebuilder:validation:MaxLength=100
+	// +optional
+	UtilizationThreshold *string `json:"utilizationThreshold,omitempty"`
```

Comment:
Have you thought about converting this to a numeric respresentation of a percentage? An int32 with a range of 0-100 valid? That might be easier to express rather than expressing it a an decimal value